### PR TITLE
Open API docs for application endpoint

### PR DESCRIPTION
**Pull request by sayantan1413**
Signed-off-by: sayantan1413 <sayantanbose2001@gmail.com>

**Description**

Added the Open API docs for application endpoint 
- /api/experimental/application/deploy.   [POST, DELETE]
- /api/experimental/application/{id}          [DELETE]

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: handlers/meshery_application_handler.go
Comment (piyushsingariya) [inline handlers/meshery_application_handler.go:189]: How can we attach a parameter as a response?

File: handlers/meshery_application_handler.go
Comment (piyushsingariya) [inline handlers/meshery_application_handler.go:26]: I am damn sure there are parameters(in body) that this handler must be expecting, because we usually attach the file in form of body with the request.

**Review by piyushsingariya — CHANGES_REQUESTED** [review]

File: handlers/meshery_application_handler.go
Comment (sayantan1413) [reply handlers/meshery_application_handler.go:189]: Oh yes I am sorry but I think I should have used `noContentWrapper` right?

File: handlers/meshery_application_handler.go
Comment (sayantan1413) [reply handlers/meshery_application_handler.go:26]: Screenshot 2021-07-25 at 5 58 04 PM
Hey @piyushsingariya does this look good to you ?

File: handlers/meshery_application_handler.go
Comment (piyushsingariya) [reply handlers/meshery_application_handler.go]: Yeah, you can add that or leave it empty.

File: handlers/meshery_application_handler.go
Comment (piyushsingariya) [reply handlers/meshery_application_handler.go]: Not sure, but could you match it to Pattern file POST method?

**Comment by sayantan1413** [comment]
Screenshot 2021-07-25 at 8 05 08 PM
Hey @piyushsingariya this is the current change that I made.

File: handlers/meshery_application_handler.go
Comment (piyushsingariya) [reply handlers/meshery_application_handler.go]: Also, add the Tag as ApplicationsAPI, the plural would be better.

Comment (sayantan1413) [reply handlers/meshery_application_handler.go]: Oh yes I will do it.

Comment (sayantan1413) [reply handlers/meshery_application_handler.go]: Pattern file POST method you mean this endpoint `api/pattern/deploy`?

Comment (sayantan1413) [reply handlers/meshery_application_handler.go]: Screenshot 2021-07-25 at 8 32 21 PM

Comment (sayantan1413) [reply handlers/meshery_application_handler.go]: This file is not accepting anything.🤔

Comment (piyushsingariya) [reply handlers/meshery_application_handler.go]: We gotta fix that in both places. Can you go through the handler for pattern deploy, We do pass a structured JSON on the deploy URL.

Comment (sayantan1413) [reply handlers/meshery_application_handler.go]: So like I need to change the `api/pattern/deploy` in the same way I did for `api/experimental/application/deploy`?

Comment (piyushsingariya) [reply handlers/meshery_application_handler.go]: You don't need to change, you need to figure out what exactly are the parameters consumed by these endpoints.

Comment (sayantan1413) [reply handlers/meshery_application_handler.go]: Hey @piyushsingariya I checked the pattern API and found that the JSON response in the Applications API is same but the PatternAPI is not taking any parameter in it so that the only difference I see over here. So I think the parameter should be removed I guess because as fas as I see the`api/experimental/application/deploy` is not expecting any parameters to be passed upon.

File: handlers/doc.go
Comment (piyushsingariya) [inline handlers/doc.go:252]: @sayantan1413 we can remove this parameter wrapper, I checked and there isn't any specified struct for parameters, so in such case we can leave it empty.

**Comment by sayantan1413** [comment]
Screenshot 2021-07-28 at 12 30 27 AM

Hey @piyushsingariya I think the deploy url's can have the yaml files attached in this way. What's your opinion on this?

File: handlers/doc.go
Comment (piyushsingariya) [inline handlers/doc.go:300]: Why not merge them together? :wink:

**Review by piyushsingariya — CHANGES_REQUESTED** [review]

File: handlers/doc.go
Comment (sayantan1413) [reply handlers/doc.go:300]: Yeah sure !😅

**Review by dhruv0000 — APPROVED** [review]

**Comment by dhruv0000** [comment]
@piyushsingariya waiting for your approval.

**Comment by sayantan1413** [comment]
Hey @piyushsingariya do we need anymore changes here?

**Comment by leecalcote** [comment]
@sayantan1413 will you resolve any open comments that have been completely addressed?

**Comment by piyushsingariya** [comment]
@sayantan1413 we fail to send a proper request via UI. see here!
image
This extra information got added with the request(the highlighted one) that is why we fail to make a proper request.

File: handlers/doc.go
Comment (sayantan1413) [reply handlers/doc.go]: Oh yeah sure I will remove it.

File: handlers/meshery_application_handler.go
Comment (sayantan1413) [reply handlers/meshery_application_handler.go]: Ok

**Comment by piyushsingariya** [comment]
@sayantan1413 Can you give us any updates on this?

**Comment by sayantan1413** [comment]
Hey @piyushsingariya I gave several attempts to solve the extra text getting appended by I could not solve this till now. I have filed an[ issue](https://github.com/go-swagger/go-swagger/issues/2615) in go-swagger how to solve this but haven't received anything from them. I am really stuck here.😬

**Review by piyushsingariya — APPROVED** [review]

**Pull request merged**
